react/src/App.tsx

```diff
@@ -81,10 +76,69 @@ function BaseApp() {
 }
 
 export default function App() {
+    const prefersDarkMode = useMediaQuery("(prefers-color-scheme: dark)");
+    console.log(prefersDarkMode);
+    const [darkMode, setDarkMode] = useState(
+        localStorage.getItem("darkMode") === null
+            ? prefersDarkMode
+            : localStorage.getItem("darkMode") === "true"
+    );
+    const globalTheme = useMemo(
+        () =>
+            createMuiTheme({
+                typography: {
+                    fontSize: 12,
+                },
+                mixins: {
+                    toolbar: {
+                        minHeight: 48,
+                    },
+                },
+                palette: {
+                    type: darkMode ? "dark" : "light",
+                    background: {
+                        default: darkMode ? "#2A2A2B" : "#fafafa",
+                    },
+                },
+                overrides: {
+                    MuiFilledInput: {
+                        input: {
+                            "&:-webkit-autofill": {
+                                WebkitBoxShadow: `0 0 0 100px ${
+                                    darkMode ? "#565656" : "transparent"
+                                } inset`,
+                                WebkitTextFillColor: darkMode ? "#fff" : "#000",
+                            },
+                        },
+                    },
+                    MuiFormLabel: darkMode
+                        ? {
+                              root: {
+                                  "&$focused": {
+                                      color: "#fff",
+                                  },
+                              },
+                          }
+                        : {},
+                },
+            }),
+        [darkMode]
+    );
+
+    useEffect(() => {
+        setDarkMode(prefersDarkMode);
+    }, [prefersDarkMode]);
```

Comment:
This will always override the user's saved preference with the system preference. You already take care of this logic when initializing the state above so this isn't needed either.